json-endpoints/class.wpcom-json-api-site-settings-endpoint.php

```diff
@@ -299,22 +298,18 @@ public function get_settings_response() {
 					$jetpack_relatedposts_options['enabled'] = true;
 				}
 
-				if ( method_exists( 'Jetpack', 'is_module_active' ) ) {
-					$jetpack_relatedposts_options[ 'enabled' ] = Jetpack::is_module_active( 'related-posts' );
-				}
+				$jetpack_relatedposts_options[ 'enabled' ] =
+					$jetpack_relatedposts_options[ 'enabled' ]
+					&& $site->is_module_active( 'related-posts' );
 
 				$jetpack_search_supported = false;
 				if ( function_exists( 'wpcom_is_jetpack_search_supported' ) ) {
 					$jetpack_search_supported = wpcom_is_jetpack_search_supported( $blog_id );
 				}
 
-				$jetpack_search_active = false;
-				if ( method_exists( 'Jetpack', 'is_module_active' ) ) {
-					$jetpack_search_active = Jetpack::is_module_active( 'search' );
-				}
-				if ( function_exists( 'is_jetpack_module_active' ) ) {
-					$jetpack_search_active = is_jetpack_module_active( 'search', $blog_id );
-				}
+				$jetpack_search_active =
+					$jetpack_search_supported
```

Comment:
This looks like it may be changing the behavior? Before, `$jetpack_search_supported` is set to false on Jetpack, but it isn't used except in the response of the API. After, it is used in `$jetpack_search_active` which used to just check if the module was active for a Jetpack site (`is_jetpack_module_active` is wpcom only).

Reply:
Right. tl;dr with these changes, it's impossible for a Jetpack site to ever be `true` for `$jetpack_search_active`

@zinigor can you take a look?

Reply:
Yes, we'll need to implement something using SAL here, good catch. I'll try to work on that today.